services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/utils/MongoDbDocumentBuilder.java

```diff
@@ -143,12 +143,7 @@ public MongoDbDocumentBuilder withDeviceFilters(final List<Filter> filters) {
 
         filters.forEach(filter -> {
             // TODO: To implement when filter values contain patterns such as * or %
-            if (FIELD_ID.equals(filter.getField())) {
-                document.put(RegistryManagementConstants.FIELD_PAYLOAD_DEVICE_ID, filter.getValue());
-            } else {
-                document.put(MongoDbDeviceRegistryUtils.FIELD_DEVICE + filter.getField().toString().replace("/", "."),
-                        filter.getValue());
-            }
+            document.put(mapDeviceField(filter.getField()), filter.getValue());
```

Comment:
How about adding an "/id" filter to the `testSearchDevicesWithMultipleFiltersSucceeds` test, so that this code path is covered by tests as well.

Reply:
I have added an`/id` filter to that test.